### PR TITLE
feat(cce): add new data_source cce_clusters

### LINKED ISSUE #1681: [CCE] Add a listing data source for all clusters
I'd like to add data source support for listing all clusters and clusters within specific enterprise project.

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #1681 

**Special notes for your reviewer**:

**Release note**:

```release-note
add new data_source cce_clusters
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/cce' TESTARGS='-run TestAccCCEClustersDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cce -v -run TestAccCCEClustersDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccCCEClustersDataSource_basic
=== PAUSE TestAccCCEClustersDataSource_basic
=== CONT  TestAccCCEClustersDataSource_basic
--- PASS: TestAccCCEClustersDataSource_basic (977.50s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cce       977.597s
```

### CODE REVIEW

File: docs/data-sources/cce_clusters.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+* `enterprise_project_id` - (Optional, String) Specifies the enterprise project ID of the cluster.
+
+* `status` - (Optional, String) Specifies the status of the cluster.
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - Indicates a data source ID in UUID format.
```
Comment (niuzhenguo) [inline docs/data-sources/cce_clusters.md]: seems not in UUID format.

Comment (Jason-Zhang9309) [reply docs/data-sources/cce_clusters.md]: fixed

File: huaweicloud/services/cce/data_source_huaweicloud_cce_clusters.go [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+		EnterpriseProjectID: config.GetEnterpriseProjectID(d),
+	}
+
+	refinedClusters, err := clusters.List(cceClient, listOpts)
+	logp.Printf("[DEBUG] Value of allClusters: %#v", refinedClusters)
+	if err != nil {
+		return fmtp.DiagErrorf("Unable to retrieve clusters: %s", err)
+	}
+
+	ids := make([]string, 0)
```
Comment (ShiChangkuo) [inline huaweicloud/services/cce/data_source_huaweicloud_cce_clusters.go]: suggest to pre-allocate the memory of ids

File: huaweicloud/services/cce/data_source_huaweicloud_cce_clusters.go [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+
+		r := clusters.GetCert(cceClient, v.Metadata.Id)
+
+		kubeConfigRaw, err := utils.JsonMarshal(r.Body)
+
+		if err != nil {
+			logp.Printf("Error marshaling r.Body: %s", err)
+		}
+
+		cluster["kube_config_raw"] = string(kubeConfigRaw)
```
Comment (ShiChangkuo) [inline huaweicloud/services/cce/data_source_huaweicloud_cce_clusters.go:282]: is it necessary to add `kube_config_raw` as it is too huge?

File: huaweicloud/services/cce/data_source_huaweicloud_cce_clusters.go (hunk 2 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/cce/data_source_huaweicloud_cce_clusters.go]: fixed

File: huaweicloud/services/cce/data_source_huaweicloud_cce_clusters.go (hunk 3 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/cce/data_source_huaweicloud_cce_clusters.go:282]: It's a useful attribute. I prefer to keep it. And we can consider a better way to save this attribute.

**Pull request merged**